wp-includes/template.php

```diff
@@ -171,3 +171,47 @@ function wp_service_worker_error_details_template( $output = '' ) {
 function wp_service_worker_error_message_placeholder() {
 	echo '<p><!--WP_SERVICE_WORKER_ERROR_MESSAGE--></p>';
 }
+
+/**
+ * Reload the offline page and check if user comes online.
+ *
+ * @since 0.7
+ */
+function wp_service_worker_offline_page_reload() {
+	if ( ! is_offline() ) {
+		return;
+	}
+	?>
+	<script>
+		/**
+		 * Listen to changes in the network state, reload when online.
+		 * This handles the case when the device is completely offline.
+		 */
+		window.addEventListener('online', () => {
+			window.location.reload();
+		});
+
+		/**
+		 * Check if the server is responding and reload the page if it is.
+		 * This handles the case when the device is online, but the server is offline or misbehaving.
+		 */
+		async function checkNetworkAndReload() {
+			try {
+				const response = await fetch('.', {method: 'HEAD'});
```

Comment:
Any reason for `.` as opposed to `location.href`? The problem with `.` is it will drop the query parameters. 

```suggestion
				const response = await fetch(location.href, {method: 'HEAD'});
```

Reply:
That's interesting. Above we are making a fetch request to base origin to check whether we are getting response OK or not. I wonder how making a fetch request to some origin can drop query params of another origin.